### PR TITLE
Gulp screenshots enhancements

### PR DESCRIPTION
**Pull request by bchiang7**
## Changes

- Updates [puppeteer](https://pptr.dev/) to latest version (8.0.0)
- Refactors some of the gulp screenshots code to start the server & browser only once to cut down on build times

## Steps to test

- Pull down this branch, `npm install`, then run `npm link`
- In another terminal tab, clone & install the [Vanderbilt repo](https://github.com/Upstatement/vanderbilt-design-system).
- Checkout the branch for [this PR](https://github.com/Upstatement/vanderbilt-design-system/pull/223)
- `npm link @upstatement/puppy`
- `npm start` and make sure there are no errors and the thumbnails show up on the homepage

## To do

- [x] Add comments

### CODE REVIEW

File: lib/gulp/screenshots/index.js [hunk 1]
```diff
@@ -24,55 +24,114 @@ const DEFAULT_OPTIONS = {
...
+
+  return captureOptions;
+}
+
+/**
+ * Start the server to visit in the headless Chrome instance
+ *
+ * @see https://nodejs.org/api/http.html#http_http_createserver_options_requestlistener
+ *
+ * @returns {Promise}
```
Comment (mgburns) [inline lib/gulp/screenshots/index.js]: `server.listen()` returns a [`Net.server`](https://nodejs.org/api/net.html#net_class_net_server) instance, not a promise:
https://nodejs.org/api/net.html#net_server_listen

**Review by mgburns — APPROVED** [review]
Flagged one misleading doc block inline, but good to go beyond that. Thanks for the enhancement! 💅

File: lib/gulp/screenshots/index.js (hunk 1 above)
Comment (bchiang7) [reply lib/gulp/screenshots/index.js]: Good catch, that was a copy paste error!

Comment (mgburns) [reply lib/gulp/screenshots/index.js]: I figured! Caught my eye because I noticed we weren't `await`ing anything at the call site.

**Pull request merged**